### DISCUSSION TITLE
How to overwrite spans of Example object?

### DISCUSSION THREAD

**Discussion by jamnicki** [explosion/spaCy#11829] [Help: Coding & Implementations] [answered]
Hello!
I need to overwrite spans list of existing `Example` object.

```python
example["doc_annotation"]["spans"][spans_key] = new_span_group
```

I encoutered following error executing above code and I have no idea of fix:

```powershell
TypeError: 'spacy.training.example.Example' object is not subscriptable
```

Do I need to create new `Example` object based on data from already existing? How can I do this?

**Comment by ljvmiranda921** [comment, marked as answer]
Hi @jamnicki , I'm curious as to why you're overwriting spans in the `Example` object, are you trying to train using a custom training loop, etc.? But to answer your question, you should only update the `reference` attribute of your [`Example` object](https://spacy.io/api/example) (`Example.reference` is a [`Doc`](https://spacy.io/api/doc)).

```python
example.reference.spans[spans_key] = span_group
```

**Reply by jamnicki** [reply]
Thank you again @ljvmiranda921!
I'm implementing active learning loop on unlabeled data. Long story short, get batch of unlabeled records basing on some informativeness metrics, annotate the data in the GUI, get annotations (there I have to do this), train, repeat 😃

Code: [here](https://github.com/jamnicki/bachelor_thesis_project/blob/master/scripts/active_learning_loop.py)
